tests/match_rules.spec.js

```diff
@@ -190,6 +191,10 @@ describe("Test matchRules", () => {
     expect(matchRules(mainSource, RULE_WITH_FUNCTION)).toBe(true);
   });
 
+  it("should execute the function with two parameters and second param when be includes source param's all keys", () => {
+    expect(matchRules(mainSource, RULE_WITH_FUNCTION)).toBe(true);
```

Comment:
This is good but we should add one more check using jest spy:
refer to the docs: 
https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname
https://codewithhugo.com/jest-fn-spyon-stub-mock/

We can spy on `RULE_WITH_FUNCTION.age` and check if has been called with the expected parameter. I.e the key value as first param and the source as the second.